thrift/src/main/java/com/linecorp/armeria/client/thrift/THttpClientDelegate.java

```diff
@@ -108,7 +109,9 @@ public RpcResponse execute(ClientRequestContext ctx, RpcRequest call) throws Exc
             }
         } catch (Throwable cause) {
             reply.completeExceptionally(cause);
-            return reply;
+            try (SafeCloseable ignored = ctx.push()) {
+                return reply;
```

Comment:
Why do we need to push the context here?

Reply:
I don't know. 😅